ovs/dal/dataobject.py

```diff
@@ -736,7 +739,7 @@ def delete(self, abandon=None, _hook=None):
                 self._persistent.apply_transaction(transaction)
                 successful = True
             except KeyNotFoundException as ex:
-                if ex.message != self._key:
+                if self._key not in ex.message:
```

Comment:
Hmm, odd, I was under the impression that ex.message is always exactly equal to the key that was not found. Can you add the example on which you based this change in a comment?